libndt.cpp

```diff
@@ -88,6 +94,9 @@ class Client::Impl {
   Socket sock = -1;
   std::vector<uint64_t> granted_suite;
   Settings settings;
+#ifdef HAVE_OPENSSL
+  std::map<Socket, SSL *> fd_to_ssl;
+#endif
```

Comment:
The actual code should be in a separate file. This should allow us to write unit tests.